src/main/java/org/embulk/output/s3/S3FileOutputPlugin.java

```diff
@@ -173,8 +180,17 @@ else if (region.isPresent()) {
                 // no explicit region or endpoint as the region (inferrable from endpoint) are necessary for signing.
                 builder.setEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration("s3.amazonaws.com", null));
             }
-
+            
             builder.withForceGlobalBucketAccessEnabled(true);
+            
+            // This code is written to ensure compatibility with bucket services like OCI (Oracle Cloud Infrastructure),
+            // which use a different endpoint format. Enabling path-style access allows the client to work with such services.
+            // Additionally, global bucket access is disabled to avoid conflicts 
+            if (path_style_access.orElse(false)) {
+                builder.withPathStyleAccessEnabled(true);
+                builder.withForceGlobalBucketAccessEnabled(false);
+            }
+
```

Comment:
Setting `withForceGlobalBucketAccessEnabled` to once `true` and then to `false` again looks redundant. Could you unify this part?

Also, I don't think we need to mention OCI specifically, too. Instead, we'll need to explain why the global bucket access needs to be disabled when the path-style access is used, with a pointer to some (official) document.